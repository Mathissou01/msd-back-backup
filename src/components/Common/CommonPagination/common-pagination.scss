@use "src/styles/02-tools/tools.typography" as typo;
@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.shadows" as shadows;

.c-CommonPagination {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;

  &__MediaPerPage {
    display: flex;
    flex-direction: row;
    gap: 16px;
    align-items: center;
    justify-content: center;
    padding: 0;
  }

  &__Select {
    @include shadows.shadow(1);

    padding: 10px;
    color: colors.$blue-1;
    background-color: #fff;
    border: hidden;
    border-radius: 5px;

    &:focus {
      outline: none;
    }
  }

  &__Label {
    color: colors.$blue-1;
  }

  &__Navigation {
    display: flex;
    flex-direction: row;
    gap: 16px;
    align-items: center;
    padding: 0;
  }

  &__NumberPage {
    display: flex;
    flex-direction: row;
    gap: 8px;
    align-items: flex-start;
    justify-content: center;
    padding: 4px 0 2px;
  }

  &__Page {
    color: colors.$blue-1;

    &_active {
      color: colors.$expert-blue;
      font-weight: 700;
    }
  }

  &__ButtonsPreview,
  &__ButtonsNext {
    display: flex;
    flex-direction: row;
    gap: 8px;
    align-items: flex-start;
    padding: 0;

    &_disabled {
      opacity: 0.3;
      user-select: none;
      pointer-events: none;
    }
  }

  &__ButtonsPreview {
    transform: rotate(180deg);
  }

  &__Icon {
    display: flex;
  }

  button {
    padding: 0;

    &:focus {
      outline: none;
    }
  }
}
