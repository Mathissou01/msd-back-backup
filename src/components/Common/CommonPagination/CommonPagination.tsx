import Image from "next/image";
import classNames from "classnames";
import React from "react";
import "./common-pagination.scss";

interface ICommonPaginationProps {
  pageCount: number | undefined;
  page: number;
  pageSize?: Array<number>;
  onPreviousPage: () => void;
  onNextPage: () => void;
  onFirstPage: () => void;
  onLastPage: () => void;
  onSpecificPage: (i: number) => void;
  setCurrentPagesize: React.Dispatch<React.SetStateAction<number>>;
}

export default function CommonPagination({
  pageCount,
  page,
  pageSize,
  onPreviousPage,
  onNextPage,
  onFirstPage,
  onLastPage,
  onSpecificPage,
  setCurrentPagesize,
}: ICommonPaginationProps) {
  const buttonPreviousClassNames = classNames(
    "c-CommonPagination__ButtonsPreview",
    {
      "c-CommonPagination__ButtonsPreview_disabled": page === 1,
    },
  );

  const buttonNextClassNames = classNames("c-CommonPagination__ButtonsNext", {
    "c-CommonPagination__ButtonsNext_disabled": page === pageCount,
  });

  function handleCurrentPage(value: number) {
    setCurrentPagesize(value);
  }

  function handlePagesLinks(pageCount: number | undefined) {
    const linksTemplate = [];
    const lastLinkIndex = pageCount || 0;

    for (let i = 1; i <= lastLinkIndex; i++) {
      linksTemplate.push(
        <button
          key={i}
          className={classNames("c-CommonPagination__Page", {
            "c-CommonPagination__Page_active": i === page,
          })}
          onClick={() => onSpecificPage(i)}
        >
          {i}
        </button>,
      );
    }

    return linksTemplate;
  }

  return (
    <div className="c-CommonPagination">
      {pageSize ? (
        <div className="c-CommonPagination__MediaPerPage">
          <select
            className="c-CommonPagination__Select"
            name="mediaPerPage"
            id="mediaPerPage"
            onChange={(t) => {
              handleCurrentPage(parseInt(t.target.value));
            }}
          >
            {pageSize?.map((size) => (
              <option key={size} value={size}>
                {size}
              </option>
            ))}
          </select>
          <span className="c-CommonPagination__Label">m√©dias par page</span>
        </div>
      ) : null}
      <nav className="c-CommonPagination__Navigation">
        <div className={buttonPreviousClassNames}>
          <button
            type="button"
            className="c-CommonPagination__Icon"
            onClick={onPreviousPage}
          >
            <Image
              src="/images/pictos/arrow-button.svg"
              alt="arrow button"
              height={24}
              width={24}
              className="c-CommonPagination__IconDisabled"
            />
          </button>
          <button
            type="button"
            className="c-CommonPagination__Icon"
            onClick={onFirstPage}
          >
            <Image
              src="/images/pictos/button.svg"
              alt="Button First Page"
              height={24}
              width={24}
              className="c-CommonPagination__IconDisabled"
            />
          </button>
        </div>
        <div className="c-CommonPagination__NumberPage">
          {handlePagesLinks(pageCount)}
        </div>
        <div className={buttonNextClassNames}>
          <button
            type="button"
            className="c-CommonPagination__Icon"
            onClick={onNextPage}
          >
            <Image
              src="/images/pictos/arrow-button.svg"
              alt="arrow button"
              height={24}
              width={24}
            />
          </button>
          <button
            type="button"
            className="c-CommonPagination__Icon"
            onClick={onLastPage}
          >
            <Image
              src="/images/pictos/button.svg"
              alt="Bouton Last page"
              height={24}
              width={24}
            />
          </button>
        </div>
      </nav>
    </div>
  );
}
