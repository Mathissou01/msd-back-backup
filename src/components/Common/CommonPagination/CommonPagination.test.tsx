import { fireEvent, render, screen } from "@testing-library/react";
import CommonPagination from "./CommonPagination";

const mock = {
  page: 1,
  pageCount: 5,
  pagiSize: [10, 20, 50, 100],
};

describe("CommonPagination", () => {
  it("renders", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <CommonPagination
        pageCount={mock.pageCount}
        page={mock.page}
        pageSize={mock.pagiSize}
        onPreviousPage={handleClick}
        onNextPage={handleClick}
        onFirstPage={handleClick}
        onLastPage={handleClick}
        onSpecificPage={handleClick}
        setCurrentPagesize={handleClick}
      />,
    );
    expect(container).toMatchSnapshot();
    fireEvent.click(screen.getByText(1));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
