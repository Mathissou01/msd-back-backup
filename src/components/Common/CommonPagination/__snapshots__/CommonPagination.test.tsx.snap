// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonPagination renders 1`] = `
<div>
  <div
    class="c-CommonPagination"
  >
    <div
      class="c-CommonPagination__MediaPerPage"
    >
      <select
        class="c-CommonPagination__Select"
        id="mediaPerPage"
        name="mediaPerPage"
      >
        <option
          value="10"
        >
          10
        </option>
        <option
          value="20"
        >
          20
        </option>
        <option
          value="50"
        >
          50
        </option>
        <option
          value="100"
        >
          100
        </option>
      </select>
      <span
        class="c-CommonPagination__Label"
      >
        m√©dias par page
      </span>
    </div>
    <nav
      class="c-CommonPagination__Navigation"
    >
      <div
        class="c-CommonPagination__ButtonsPreview c-CommonPagination__ButtonsPreview_disabled"
      >
        <button
          class="c-CommonPagination__Icon"
          type="button"
        >
          <img
            alt="arrow button"
            class="c-CommonPagination__IconDisabled"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/images/pictos/arrow-button.svg"
            style="color: transparent;"
            width="24"
          />
        </button>
        <button
          class="c-CommonPagination__Icon"
          type="button"
        >
          <img
            alt="Button First Page"
            class="c-CommonPagination__IconDisabled"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/images/pictos/button.svg"
            style="color: transparent;"
            width="24"
          />
        </button>
      </div>
      <div
        class="c-CommonPagination__NumberPage"
      >
        <button
          class="c-CommonPagination__Page c-CommonPagination__Page_active"
        >
          1
        </button>
        <button
          class="c-CommonPagination__Page"
        >
          2
        </button>
        <button
          class="c-CommonPagination__Page"
        >
          3
        </button>
        <button
          class="c-CommonPagination__Page"
        >
          4
        </button>
        <button
          class="c-CommonPagination__Page"
        >
          5
        </button>
      </div>
      <div
        class="c-CommonPagination__ButtonsNext"
      >
        <button
          class="c-CommonPagination__Icon"
          type="button"
        >
          <img
            alt="arrow button"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/images/pictos/arrow-button.svg"
            style="color: transparent;"
            width="24"
          />
        </button>
        <button
          class="c-CommonPagination__Icon"
          type="button"
        >
          <img
            alt="Bouton Last page"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/images/pictos/button.svg"
            style="color: transparent;"
            width="24"
          />
        </button>
      </div>
    </nav>
  </div>
</div>
`;
